src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -311,13 +310,34 @@ class LspConnectionManager {
                     * buffer of executeCommands, send to Lsp at a throttled pace (hack timeouts)  
                   * longterm: middleware? changes in Lsp?
     */
```

Comment:
Should this comment also be updated since each time the the buffer updates, the entire composition is reinitiated?

Reply:
we still use the buffer, because it does:
* re-makes the entire composition on re-sync (the change here)
* still buffers user click-by-click for all other actions.
   * these actions are not a re-init.

in most cases, messages will be inserted into the buffer and then immediately incremented. Exceptions being:
* the user is clicking faster than the roundtrip request/response with the LSP
* or we still in the page (re-)load process


I updated the checklist under the description here, to make evident what still remains unfixed. https://github.com/influxdata/ui/issues/5575 The question is how frequently we'll see users getting out-of-sync; thereby setting the urgency for the middleware.